typing/env.ml

```diff
@@ -922,7 +916,8 @@ let rec find_module_descr path env =
     Pident id ->
       begin match IdTbl.find_same id env.components with
       | Value x -> fst x
-      | Persistent -> (find_pers_struct (Ident.name id)).ps_comps
+      | Persistent | exception Not_found ->
+          (find_pers_struct (Ident.name id)).ps_comps
```

Comment:
In the previous version of this code (pre #2041), there was the following check before calling `find_pers_struct`:
```ocaml
if Ident.persistent id && not (Ident.name id = !current_unit)
```

I haven't read #2041 yet, so it's might be obvious from there, but can you explain why it is it not needed here (since you're reintroducing the call to `find_pers_struct`)?
Intuitively, it looks like it should be reintroduced as well.

Reply:
TBH, I'm not sure why it was needed in the first place. I can't remember either why it was removed in #2041. I'm happy to re-introduce it if you think it's necessary.

Reply:
I think you should reintroduce it.

I guess that they were removed in #2041 because the `exception Not_found` branch wasn't handled (and in the `Persistent` case, well, you already knew had that information).

I don't think you'd be able to observe the impact of the first part of the check (the `Ident.persistent`) unless there was a bug in the compiler, but I think it's good to have.

As for the second part of the check, it was introduced to handle [MPR#6886](https://caml.inria.fr/mantis/view.php?id=6886), which I'd guess you can reproduce with the code as currently proposed. (It'd be good to add this example to the testsuite!)

Reply:
> I think you should reintroduce it.

To make that clearer: you should reintroduce it here and in the other functions with that same code pattern.

Reply:
Ok. Should I put this check only on the `exception Not_found` branch or on `Persistent | exception Not_found`?

Reply:
Well, looking at the code of your previous PR, it seems that the only place where `Persistent` is added to the `components` table is in `add_persistent_structure`, where you already perform that exact check.
So I guess the answer is "it is only needed in the exceptional branch".

Reply:
I like the change, thanks!